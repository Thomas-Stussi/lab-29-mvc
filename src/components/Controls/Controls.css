.Controls{
  display: grid;
  grid-template-columns: 1fr;
  width: 100%;
  border: 1px solid black;
  margin: 2.5px;
}
