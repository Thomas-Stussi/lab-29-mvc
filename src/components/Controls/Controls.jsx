/* eslint-disable max-len */
import React from 'react';
import PropTypes from 'prop-types';
import styles from './Controls.css';

const Controls = ({ url, method, body, onChange, onSubmit }) => {
  return (
    <section>
      <form onSubmit={onSubmit} className={styles.Controls}>
        <input type="text" name="url" placeholder="URL" value={url} onChange={onChange} />
        <section>

          <div>
            <input type="radio" id="get" name="method" value="GET" checked={method === 'GET'} onChange={onChange} />
            <label htmlFor="get">GET</label>

            <input type="radio" id="post" name="method" value="POST" checked={method === 'POST'} onChange={onChange} />
            <label htmlFor="post">POST</label>

            <input type="radio" id="put" name="method" value="PUT" checked={method === 'PUT'} onChange={onChange} />
            <label htmlFor="put">PUT</label>

            <input type="radio" id="patch" name="method" value="PATCH" checked={method === 'PATCH'} onChange={onChange} />
            <label htmlFor="patch">PATCH</label>

            <input type="radio" id="delete" name="method" value="DELETE" checked={method === 'DELETE'} onChange={onChange} />
            <label htmlFor="delete">DELETE</label>
          </div>
          <textarea placeholder="Raw JSON Body" name="body" value={body} onChange={onChange}></textarea>
          <br/>
          <button data-testid="button">Send Request</button>
        </section>
      </form>
    </section>
  );
};

Controls.propTypes = {
  url: PropTypes.string.isRequired,
  method: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired
};

export default Controls;
