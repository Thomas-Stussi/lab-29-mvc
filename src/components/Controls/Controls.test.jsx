/* eslint-disable max-len */
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Controls from './Controls';

describe('Controls component', () => {
  afterEach(() => cleanup());
  it('render Controls component', () => {
    const { asFragment } = render(<Controls
      url="https://www.thecocktaildb.com/api/json/v1/1/random.php"
      method="GET"
      onSubmit={() => { }}
      onChange={() => { }}
    />);

    expect(asFragment()).toMatchSnapshot();
  });
});
