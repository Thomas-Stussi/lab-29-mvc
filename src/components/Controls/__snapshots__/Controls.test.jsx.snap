// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls component render Controls component 1`] = `
<DocumentFragment>
  <section>
    <form
      class="Controls"
    >
      <input
        name="url"
        placeholder="URL"
        type="text"
        value="https://www.thecocktaildb.com/api/json/v1/1/random.php"
      />
      <section>
        <div>
          <input
            checked=""
            id="get"
            name="method"
            type="radio"
            value="GET"
          />
          <label
            for="get"
          >
            GET
          </label>
          <input
            id="post"
            name="method"
            type="radio"
            value="POST"
          />
          <label
            for="post"
          >
            POST
          </label>
          <input
            id="put"
            name="method"
            type="radio"
            value="PUT"
          />
          <label
            for="put"
          >
            PUT
          </label>
          <input
            id="patch"
            name="method"
            type="radio"
            value="PATCH"
          />
          <label
            for="patch"
          >
            PATCH
          </label>
          <input
            id="delete"
            name="method"
            type="radio"
            value="DELETE"
          />
          <label
            for="delete"
          >
            DELETE
          </label>
        </div>
        <textarea
          name="body"
          placeholder="Raw JSON Body"
        />
        <br />
        <button
          data-testid="button"
        >
          Send Request
        </button>
      </section>
    </form>
  </section>
</DocumentFragment>
`;
